tests/test_adjoint_jacobian.py

```diff
@@ -968,8 +968,8 @@ def convert_to_array_lightning(params):
         (qml.Hamiltonian([1], [qml.PauliZ(0)]), HamiltonianKokkos_C64, HamiltonianKokkos_C128),
         (
             qml.PauliZ(0) @ qml.Hadamard(1) @ (0.1 * (qml.PauliZ(2) + qml.PauliX(3))),
-            TensorProdObsKokkos_C64,
-            TensorProdObsKokkos_C128,
+            HamiltonianKokkos_C64,
+            HamiltonianKokkos_C128,
```

Comment:
Why does that change?

Reply:
This was part of the operator promotion discussion yesterday. For the new operators involving sums and prods, this will always be Hamiltonian-like. For the old syntax, this gave us a tensor like, rather than a Hamiltonian. A recent PR to the PL core made the change to ensure this is promoted to a Hamiltonian as:
```
qml.PauliZ(0) @ qml.PauliY(1) @ (0.5*qml.PauliX(2) + 1.7* qml.PauliY(3)) # Tensor
-> 
(0.5*qml.PauliZ(0) @ qml.PauliY(1) @ qml.PauliX(2) + 1.7*qml.PauliZ(0) @ qml.PauliY(1) @ qml.PauliY(3)) # Hamiltonian
```